addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -71,6 +71,11 @@ class MessageList extends Component {
 
     mounted() {
         this._update();
+        if (this.threadViewer.threadCacheInitialScrollPosition !== undefined) {
+            if (this.props.hasScrollAdjust) {
+                this.el.scrollTop = this.threadViewer.threadCacheInitialScrollPosition;
+            }
+        }
```

Comment:
I would not put this specifically in `mounted()`, because this makes this behaviour dependent on this component lifecycle. And we want to avoid that, in order to have more declarative components (less errors, and would help migrate to functional components with OWL 2.0).

Think would have to use a hint somehow, I would say `componentHint` so that it's managed as a scroll adjustment like others.

Reply:
I'm not sure to understand when this componentHint should be fired ?

Reply:
It saves scroll positions on `will_<show/hide>_home_menu` and it recovers scroll positions on `<show/hide>_home_menu`. So I would expect firing hint as a consequence of non-`will_[...]` events.